### ISSUE TITLE
GPU test is failing consistently on release-0.4 branch

### ISSUE THREAD

**Issue by CecileRobertMichon** [kubernetes-sigs/cluster-api-provider-azure#1354]
/kind failing-test

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#capz-periodic-capi-e2e-full-release-0-4

```
/home/prow/go/src/sigs.k8s.io/cluster-api-provider-azure/test/e2e/azure_test.go:314
Timed out after 300.046s.
Job default/cuda-vector-add failed
```

It might be an issue with the job, or with the GPU nodes themselves. #1344 recently updated the job image in the main branch, backporting the change might be enough to fix it (or it might be a bigger issue). Note that the GPU template is different in release-0.4 as it does not use the nvidia operator.

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

**Comment by mboersma** [comment]
I think a cherry-pick of #1693 might fix this. The failures I looked at in the 0.4 branch seemed to be Job timeouts.

**Comment by mboersma** [comment]
This was fixed by #1705–*for a time*. It is [broken again](https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#capz-periodic-capi-e2e-full-release-0-4) consistently since 9/27. I will investigate as soon as time permits.
